application/src/main/java/org/togetherjava/tjbot/commands/utils/MessageUtils.java

```diff
@@ -56,4 +59,27 @@ public static String escapeMarkdown(String text) {
         return afterEscape.replace("\\```", "\\`\\`\\`");
     }
 
+    /**
+     * Converts a slash command text to a mentioned slash command, which you can directly click on
+     * in Discord
+     *
+     * @param guild guild that owns the command
+     * @param command command name
+     * @return Formatted string for mentioned slash command
+     * @throws IllegalArgumentException when the command name doesn't match with the guild's
+     *         commands
+     */
+    public static String mentionSlashCommand(Guild guild, String command) {
+        AtomicLong commandId = new AtomicLong();
+        guild.retrieveCommands()
+            .queue(commands -> commands.stream()
+                .filter(c -> c.getName().equalsIgnoreCase(command))
+                .findFirst()
+                .ifPresentOrElse(c -> commandId.set(c.getIdLong()), () -> {
+                    throw new IllegalArgumentException("Command %s does not exist in guild %s"
+                        .formatted(command, guild.getId()));
+                }));
+        return String.format("</%s:%d>", command, commandId.get());
```

Comment:
I do see one problem with this design, which we should address. The method makes an actual request to JDA/Discord, but blocks on it.

We should avoid blocking and instead return a `RestAction<String>` out of this. I know that this will uglify the call-sites and make it more complex, but we dont have much choice here.

Reply:
This doesn't block, this code actually simply doesn't work.
Queue runs after the `return`  statement

But yes, rewriting it so it returns a `RestAction<String>`  would be better

Reply:
Made a change to return a `RestAction<String>` as suggested, but then how should I handle it on the caller side ? Right now I've used `MessageUtils.mentionSlashCommand(...).complete()` but it makes it blocking on the caller's side instead of inside the method, so I'm quite unsure this is what you expect.

Reply:
ur supposed to have a single chain. all actions will be mapped on it. at the end of everything, there will be a `queue()` at some point, which sends out the whole chain of actions.